src/nlsolve/anderson.jl

```diff
@@ -108,20 +112,19 @@ function (S::NLAnderson{true})(integrator)
     mul!(ztmp, mass_matrix, z)
     @. z₊ -= ztmp
   end
-  gs[:,1] .= vec(z₊)
+  gs[1] .= vec(z₊)
   @. dz = z₊ - z
   ndz = integrator.opts.internalnorm(dz)
   for t = (S.n+1):-1:2
-    @. zs[:,t] = zs[:,t-1]
-    @. gs[:,t] = gs[:,t-1]
+    zs[t] = zs[t-1]
```

Comment:
.= otherwise these now alias

Reply:
If you want to use pointer swaps, check out this https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/pull/602/files#diff-a47f650c5d66e06a667d8a85cec5c42eR865

Reply:
The issue is that the top one doesn't become the bottom one, so after enough steps they all are the same array.

Reply:
Oh got it. Thanks!